python/mxnet/gluon/trainer.py

```diff
@@ -255,6 +255,13 @@ def learning_rate(self):
         else:
             return self._optimizer.learning_rate
 
+    @property
+    def optimizer(self):
```

Comment:
I m exposing optimizer through trainer, as estimator only has access to trainer. For now it's only used for logging and type and name. Let me know if I should remove this.